src/main/java/org/opensearch/neuralsearch/query/NeuralSparseQueryBuilder.java

```diff
@@ -376,16 +376,24 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
     }
 
     private static void validateForRewrite(String queryText, String modelId) {
-        if (StringUtils.isBlank(queryText) || StringUtils.isBlank(modelId)) {
+        if (StringUtils.isBlank(modelId) && !isClusterOnOrAfterMinReqVersionForDefaultModelIdSupport()) {
```

Comment:
`isClusterOnOrAfterMinReqVersionForDefaultModelIdSupport` only checks whether the default model id feature is supported in the current cluster by `OpenSearch version`. Have you checked that default model id has been configured?